lib/cext/patches/racc_patches.rb

```diff
@@ -0,0 +1,123 @@
+            match: 'static VALUE reduce0 _((VALUE block_args, VALUE data, VALUE self));',
+            replacement: 'static VALUE reduce0 _((VALUE block_args, VALUE data));'
+        },
+        { # Change signature because when we call the callback we're not providing a self value -- difference from MRI.
+            match: 'reduce0(VALUE val, VALUE data, VALUE self)',
+            replacement: 'reduce0(VALUE val, VALUE data)'
+        },
+        { # We don't guarantee we #define macros, but racc has backwards-compatiblity code to check for them. Undo racc's work to avoid bad redefinitions of our implementations.
+            match: 'define ID2SYM(i) ULONG2NUM(i)',
+            replacement: 'define ID2SYM_DEAD_CODE(i) ULONG2NUM(i)'
+        },
+        { # We don't guarantee we #define macros, but racc has backwards-compatiblity code to check for them. Undo racc's work to avoid bad redefinitions of our implementations.
+            match: 'define SYM2ID(v) ((ID)NUM2ULONG(v))',
+            replacement: 'define SYM2ID_DEAD_CODE(v) ((ID)NUM2ULONG(v))'
+        },
+        { # We don't guarantee we #define macros, but racc has backwards-compatiblity code to check for them. Undo racc's work to avoid bad redefinitions of our implementations.
+            match: 'define SYMBOL_P(v) FIXNUM_P(v)',
+            replacement: 'define SYMBOL_P_DEAD_CODE(v) FIXNUM_P(v)'
+        },
+        { # We don't guarantee we #define macros, but racc has backwards-compatiblity code to check for them. Undo racc's work to avoid bad redefinitions of our implementations.
+            match: 'define LONG2NUM(i) INT2NUM(i)',
+            replacement: 'define LONG2NUM_DEAD_CODE(i) INT2NUM(i)'
+        },
+        { # racc_cparse and racc_yyparse
+          match: 'volatile VALUE',
+          replacement: 'VALUE'
+        },
+        { # struct cparse_params. Creating a handle for a primitive value doesn't seem to work, so convert the `Fixnum` field to an `int` to avoid the need for a handle.
+          match: 'VALUE t;',
+          replacement: 'int t;'
+        },
+        { # initialize_params
+          match: 'v->t = INT2FIX(FINAL_TOKEN + 1);',
+          replacement: 'v->t = FINAL_TOKEN + 1;'
+        },
+        { # parse_main
+          match: 'NUM2LONG(v->t)',
+          replacement: 'v->t'
+        },
+        { # extract_user_token
+          match: '*tok = AREF(block_args, 0);',
+          replacement: '*tok = rb_tr_handle_for_managed_leaking(AREF(block_args, 0));'
+        },
+        { # extract_user_token
+          match: '*val = AREF(block_args, 1);',
+          replacement: '*val = rb_tr_handle_for_managed_leaking(AREF(block_args, 1));'
+        },
+        { # parse_main
+          match: 'v->t = rb_hash_aref(v->token_table, tok);',
+          replacement: 'VALUE tr_tmp = rb_hash_aref(v->token_table, tok == 0 ? Qfalse : rb_tr_managed_from_handle(tok));'
```

Comment:
`v->t` has been patched from a `VALUE` to an `int`. The only case it would not be a `Fixnum` without patching is this one assignment where it could be `nil`. I'd rather we didn't have to change the type of `v->t`, but I couldn't get it working.